### ISSUE TITLE
Unable to convert Numpy array to Tensor after keras.sequence.pad_sequences()

### ISSUE THREAD

**Issue by aleSuglia** [pytorch/pytorch#3513]
Hello,

I'm using the ``pad_sequences`` function that is present in the Keras library in order to get back a padded numpy array. If you try to transform the resulting array in a LongTensor it will fail. 

```
In [42]: x
Out[42]: [[1, 2], [1, 2, 3, 4], [2, 3, 4]]

In [43]: a = keras.preprocessing.sequence.pad_sequences(x, padding="post", maxle
    ...: n=4)

In [44]: a
Out[44]:
array([[1, 2, 0, 0],
       [1, 2, 3, 4],
       [2, 3, 4, 0]], dtype=int32)

In [45]: torch.LongTensor(a)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-45-25df5cb2e023> in <module>()
----> 1 torch.LongTensor(a)

RuntimeError: tried to construct a tensor from a nested int sequence, but found an item of type numpy.int32 at index (0, 0)
```

If you transform the array in a list again the code works. Is it a bug?

I am using the version '0.2.0_3' of PyTorch on a MacOS High Sierra.

Cheers,
Alessandro

**Comment by apaszke** [comment]
That's weird. As a workaround `torch.from_numpy` should work

**Comment by bheinzerling** [comment]
This is not caused by `pad_sequences`. The culprit is the numpy array dtype:

```Python
>>> torch.LongTensor(np.arange(10, dtype="int32"))
RuntimeError: tried to construct a tensor from a int sequence, but found an item of type numpy.int32 at index (0)
>>> torch.LongTensor(np.arange(10, dtype="int64"))  # works
```

Neither of these work in torch.cuda, btw:

```Python
>>> torch.cuda.LongTensor(np.arange(10, dtype="int32"))
RuntimeError: tried to construct a tensor from a int sequence, but found an item of type numpy.int32 at index (0)
>>> torch.cuda.LongTensor(np.arange(10, dtype="int64")
RuntimeError: tried to construct a tensor from a int sequence, but found an item of type numpy.int64 at index (0)
```

With float it looks like this on my system:
```Python
torch.FloatTensor(np.arange(10, dtype="float64"))  # works
torch.FloatTensor(np.arange(10, dtype="float32"))  # works
torch.cuda.FloatTensor(np.arange(10, dtype="float32"))  # exception
torch.cuda.FloatTensor(np.arange(10, dtype="float64"))  # works
```

Calling `tolist()` on the numpy array or `torch.from_numpy` works in all cases.

**Comment by soumith** [comment]
fixed via #2461
